_posts/2022-9-5-Diplomat @ Google.md

```diff
@@ -12,46 +12,64 @@ FFI tool, [Diplomat](https://github.com/rust-diplomat/diplomat).
 
 # Diplomat: FFI bindings to anything
 
-Diplomat is an open source project started at Google, and was motivated by the
-lack of good FFI tools that satified ICU4X's requirements. This post focuses on
-how it works, but details on the motivation can be found in the [design document](https://github.com/rust-diplomat/diplomat/blob/main/docs/design_doc.md).
+Diplomat is an open source project started by the ICU4X project, designed by
+Manish, and originally implemented by Shadaj, a past Google intern, and was
+motivated by the lack of good FFI tools that satified ICU4X's requirements. This
+post focuses on how it works, but details on the motivation can be found in the
+[design document](https://github.com/rust-diplomat/diplomat/blob/main/docs/design_doc.md).
 
-Diplomat is a unidirectional FFI tool that allows programs in other languages
+Diplomat is a unidirectional[^1] FFI tool that allows programs in other languages
 to call Rust libraries via automatically generated, idiomatic APIs. At its core
 lays the abstract representation of types and methods, which represents the
 lowest common denominator for what types and methods look like across most
 modern programming languages. Since the abstract representation is so general,
 Diplomat is theoretically capable of generating bindings to any language that
-Rust has FFI support for, namely JavaScript via WebAssembly, and languages that
-support C FFI. Currently, it has support for generating bindings to C, C++, C#,
-and JavaScript/TypeScript.  Diplomat also generates native-feeling types and
+can support binding to C or WASM, since it generates `extern "C"` functions for
+each method. Currently, it has support for generating bindings to C, C++, C#,
+and JavaScript/TypeScript. Diplomat also generates native-feeling types and
 methods in the target language that call into the raw WASM/C interface.
 
-The cost of creating bindings to all these languages is just one set of FFI
-bindings, which are exposed through a module marked with the
-`#[diplomat::bridge]` proc macro. So if you have some large Rust library like
-ICU4X, you can attach Diplomat bindings and get native feeling APIs in every
-supported language at no extra cost.
+[^1]: Unidirectional FFI is when one language can invoke functions in another
```

Comment:
I'd phrase this more as: when code from one language needs to know about APIs from another language so it can call them, but not vice versa. And mention that it's most common when trying to expose a _library_ to other languages.

Part of the distinction is that callbacks can still be unidirectional, even if they're functions being called the other way.